pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -330,6 +331,9 @@ stdenv.mkDerivation ({
         export NIX_${ghcCommandCaps}="${ghcEnv}/bin/${ghcCommand}"
         export NIX_${ghcCommandCaps}PKG="${ghcEnv}/bin/${ghcCommand}-pkg"
         export NIX_${ghcCommandCaps}_DOCDIR="${ghcEnv}/share/doc/ghc/html"
+        export LD_LIBRARY_PATH="''${LD_LIBRARY_PATH:+''${LD_LIBRARY_PATH}:}${
+          concatStringsSep ":" (map (x: x + "/lib") (filter isDerivation systemBuildInputs))
```

Comment:
1. Great, I've updated the PR!
2. I added that because without it I got an "cannot coerce null to a string" error. I did not investigate further, but I suspect in my case the vulkan package is the culprit. See in `hackage-packages.nix`:
```
"vulkan" = callPackage
  ({ mkDerivation, base, vector-sized, vulkan }:
    mkDerivation { .. librarySystemDepends = [ vulkan ]; .. }
  }) {vulkan = null;};
```
Actually that is not uncommon:
```
$ grep '= null;' hackage-packages.nix | wc -l
289
```
Arguably we could filter `allBuildInputs` or `propagatedBuildInputs` and `otherBuildInputs` instead. I'll change the PR if you want, but that will result in a mass rebuild. We could also filter for not null instead of derivation, if that makes the intent clearer.